pyteal/ast/subroutine.py

```diff
@@ -304,6 +311,10 @@ def _validate_annotation(
         )
 
     def get_declaration(self) -> "SubroutineDeclaration":
+        warnings.warn(
+            "`get_declaration` is being deprecated: Please use `get_declaration_by_option` instead.",
+            DeprecationWarning,
+        )
         return self.declarations.get_declaration()
```

Comment:
@ahangsu Minor:  Since `self.declarations.get_declaration()` is deprecated, I'm wondering if it's possible to change the implementation to a non-deprecated variant?

I think `return self.get_declaration_by_option()` is also a possible option here.

```suggestion
        return self.declarations.get_declaration_by_option(False)
```

Reply:
good point, I take the change, or API caller will take double warning.